jersey-servers/src/main/java/com/palantir/remoting3/servers/jersey/TraceEnrichingFilter.java

```diff
@@ -28,17 +28,27 @@
 import javax.ws.rs.container.ContainerRequestFilter;
 import javax.ws.rs.container.ContainerResponseContext;
 import javax.ws.rs.container.ContainerResponseFilter;
+import javax.ws.rs.core.Context;
 import javax.ws.rs.core.MultivaluedMap;
 import javax.ws.rs.ext.Provider;
+import org.glassfish.jersey.server.ExtendedUriInfo;
 
 @Provider
 public final class TraceEnrichingFilter implements ContainerRequestFilter, ContainerResponseFilter {
     public static final TraceEnrichingFilter INSTANCE = new TraceEnrichingFilter();
 
+    @Context
+    private ExtendedUriInfo uriInfo;
+
     // Handles incoming request
     @Override
     public void filter(ContainerRequestContext requestContext) throws IOException {
-        String operation = requestContext.getMethod() + " /" + requestContext.getUriInfo().getPath();
+        String path = Optional.ofNullable(uriInfo)
+                .flatMap(info -> Optional.ofNullable(info.getMatchedModelResource()))
+                .flatMap(resource -> Optional.ofNullable(resource.getPath()))
```

Comment:
You can just use `Optional#map`. It returns an empty optional when the mapping function returns null.

Reply:
nice